src/test/java/org/opensearch/knn/plugin/script/KNNScriptScoringIT.java

```diff
@@ -35,12 +38,29 @@
 import static org.hamcrest.Matchers.containsString;
 
 public class KNNScriptScoringIT extends KNNRestTestCase {
+    private void randomCreateKNNIndex() throws IOException {
```

Comment:
move this function after all the public functions.

Also, why we are creating KNNIndex randomly?

Reply:
I want to test different vector values by creating native knn index or Lucene knn index with/without doc values randomly

Reply:
this makes the test little unpredictable. Can we have separate tests for native engine and Lucene. In that sense we will know if test fails for which engine it is failing.

Reply:
I tend to keep this for two reasons: 1. the output log can help us reproduce the error with a specific random seed if the test fails; 2. using randomization to test different code paths is a common approach in OpenSearch, which can make test code shorter and clearer.

Reply:
Randomization is good in data generation, but as part of UTs and ITs its better to test the concrete engine here. Because when index creation config is random we can never say that the script score is tested for nmslib, lucene and faiss engine. What if in all the CIs and build process nmslib engine never got hit and there is bug that get out in the release. Hence I wouldn't recommend creating index mappings at random here. 


Also, in this case I would not even generate random data for script score for ITs because we should validate the top results coming out of script score query to ensure that new DocValues are working perfectly.